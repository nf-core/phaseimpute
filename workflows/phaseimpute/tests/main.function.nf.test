
nextflow_function {

    name "Test Functions"
    script "../main.nf"
    tag 'workflows'
    tag 'phaseimpute'

    test("Test Function export_csv") {

        function "export_csv"

        when {
            params {
                outdir = "results"
            }
            function {
                """
                // define inputs of the function here. Example:
                input[0] = Channel.of( [ [id:'test'], file('vcf/test.vcf'), file('index/test.vcf.idx') ] )
                input[1] = ["id"]
                input[2] = [1:"vcf",2:"index"]
                input[3] = "id,vcf,index"
                input[4] = "impute.csv"
                input[5] = "imputation/csv"
                """
            }
        }

        then {
            assertAll(
                { assert function.success },
                { assert snapshot(function.result).match() }
            )
        }
    }
}
