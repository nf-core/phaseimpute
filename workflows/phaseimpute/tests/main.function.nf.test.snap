{
    "Test Function export_csv": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-03T14:25:00.306786423"
    }
}